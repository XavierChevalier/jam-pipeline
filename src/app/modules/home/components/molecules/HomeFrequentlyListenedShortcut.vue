<script setup lang="ts">
  import { string } from 'yup'
  import { isPropertyValid } from '@/app/tools/component-properties/PropertyValidator'

  defineProps({
    imageUrl: {
      type: String,
      required: true,
      validator: isPropertyValid(string().required().url()),
    },
  })
</script>

<template>
  <div class="flex items-center bg-white rounded pr-2">
    <img class="rounded" :src="imageUrl" alt="Song or album image" />
    <span class="ml-3">
      <slot />
    </span>
  </div>
</template>
