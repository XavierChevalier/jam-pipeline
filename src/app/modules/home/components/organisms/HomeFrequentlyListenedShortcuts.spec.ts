import { StorybookTestsGenerator } from '@/tests/tests-generators/StorybookTestsGenerator'
import * as Stories from '@/app/modules/home/components/molecules/HomeFrequentlyListenedShortcut.stories'

describe('HomeFrequentlyListenedShortcuts', () => {
  StorybookTestsGenerator.fromStoriesExports(Stories)
    .snapshotEachStories()
    .getComponent()
})
