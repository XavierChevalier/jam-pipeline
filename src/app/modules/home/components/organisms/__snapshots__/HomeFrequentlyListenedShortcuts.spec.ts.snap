// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeFrequentlyListenedShortcuts stories snapshots [snapshot] should render the "Default" component story correctly 1`] = `
<div
  class="flex items-center bg-white rounded pr-2"
>
  <img
    alt="Song or album image"
    class="rounded"
    src="https://picsum.photos/50"
  />
  <span
    class="ml-3"
  >
    
    Liked songs
    
  </span>
</div>
`;
